src/components/PromotionList.tsx

```diff
@@ -36,12 +36,17 @@ export default function PromotionList({
    */
   useEffect(() => {
     dispatch({ type: DispatchAction.DATA_LOADING });
```

Comment:
Does this dispatch action need to be outside of the useEffect? Or do we need another one of these dispatch actions in the useEffect for the search query

Reply:
I think I put this inside the useEffect to know when the promotions are being fetched (and in the future this would be used to determine when a Loading indicator would be displayed). It isn't being used right now but I thought I'd set it up anyways, so yeah maybe if you can add one to the if-block for the search query that would be nice!